index.js

```diff
@@ -764,22 +800,42 @@ Jimp.prototype.autocrop = function (cb) {
     }
 
     east: // east side (scan columns from east to west)
-    for (var x = 0; x < w; x++) {
-        for (var y = 0; y < h; y++) {
-            if (this.getPixelColor(x, y) !== color) {
-                // this pixel is not the same color as the first one: abort this side scan
+    for (var x = 0; x < w - minPixelsPerSide; x++) {
+        for (var y = 0 + northPixelsToCrop; y < h; y++) {
+            var colorXY = this.getPixelColor(x, y);
+            var rgba2 = Jimp.intToRGBA(colorXY);
+            var difference =
+                Math.abs(
+                    Math.max((rgba1.r - rgba2.r) ^ 2, (rgba1.r - rgba2.r - rgba1.a + rgba2.a) ^ 2) +
+                    Math.max((rgba1.g - rgba2.g) ^ 2, (rgba1.g - rgba2.g - rgba1.a + rgba2.a) ^ 2) +
+                    Math.max((rgba1.b - rgba2.b) ^ 2, (rgba1.b - rgba2.b - rgba1.a + rgba2.a) ^ 2)
+                ) / (256 * 256 * 3) 
+            ;
+            if (difference > tolerance) {
+                // this pixel is too distant from the first one: abort this side scan
                 break east;
             }
         }
         // this column contains all pixels with the same color: increment this side pixels to crop
         eastPixelsToCrop++;
     }
 
+    colorTarget = this.getPixelColor(w - 1, h - 1); // bottom right pixel color is the target color
```

Comment:
Here's an error. Bottom right pixel is being read, but the value is never used to update `rgba1` variable, and every pixel still gets compared against the top left one.